spock-core/src/main/java/org/spockframework/util/JUnit4LegacyUtil.java

```diff
@@ -0,0 +1,33 @@
+package org.spockframework.util;
+
+import java.lang.annotation.Annotation;
+
+/**
+ * The utility class to deal with the JUnit 4 annotations in Spock in the way which allows to use Spock 2
+ * without junit4.jar at runtime.
+ *
+ * @since 3.0.0
+ */
+@SuppressWarnings("unchecked")
+public class JUnit4LegacyUtil {
```

Comment:
If we move the the setup- stuff to spock-junit4 then this util doesn't need to stay in core.

Reply:
It will not be needed at all, spock-junit4 has compile time access to JUnit 4's annotations